### PR TITLE
currency_rate_update: Add support update from Russian National Bank

### PR DESCRIPTION
**Pull request by barachka**
Add source Russian National Bank for import currency rate

### CODE REVIEW

File: currency_rate_update/services/update_service_CBRF.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+from .currency_getter_interface import Currency_getter_interface
+
+class CBRF_getter(Currency_getter_interface) :
+    """Implementation of Currency_getter_factory interface
+    for The Central Bank of the Russian Federation service"""
+        
+    def get_updated_currency(self, currency_array, main_currency, max_delta_days):
+        """implementation of abstract method of Currency_getter_interface"""
+        from xmltodict import parse
+        import requests
```
Comment (guewen) [inline currency_rate_update/services/update_service_CBRF.py]: Can you import them at the top of the module?

**Comment by feketemihai** [comment]
I think for more ease to rename the class as country code_name of the bank_getter...it's the same for the rest odf national banks.

File: currency_rate_update/services/update_service_CBRF.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+from .currency_getter_interface import Currency_getter_interface
+
+class CBRF_getter(Currency_getter_interface) :
+    """Implementation of Currency_getter_factory interface
+    for The Central Bank of the Russian Federation service"""
+        
+    def get_updated_currency(self, currency_array, main_currency, max_delta_days):
+        """implementation of abstract method of Currency_getter_interface"""
+        from xmltodict import parse
```
Comment (guewen) [inline currency_rate_update/services/update_service_CBRF.py]: This is a new dependency. Do we really want to add a new lib in this module? I mean, xml can be read with lxml as well.

File: currency_rate_update/services/update_service_CBRF.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from .currency_getter_interface import Currency_getter_interface
+
+class CBRF_getter(Currency_getter_interface) :
```
Comment (feketemihai) [inline currency_rate_update/services/update_service_CBRF.py]: i think you can call it RF_CBRF_getter.

File: currency_rate_update/model/currency_rate_update.py [hunk 4]
```diff
@@ -130,6 +130,7 @@
     'CH_ADMIN_getter': CH_ADMIN_supported_currency_array,
     'MX_BdM_getter': MX_BdM_supported_currency_array,
     'PL_NBP_getter': PL_NBP_supported_currency_array,
+    'CBRF_getter': supported_currency_array,
```
Comment (feketemihai) [inline currency_rate_update/model/currency_rate_update.py]: I think you should add supported curencies.

Comment (barachka) [inline currency_rate_update/model/currency_rate_update.py]: Add supported currencies.

**Comment by barachka** [comment]
Rename getter to format country_name.
Refract code.

**Comment by barachka** [comment]
Please, review and merge request.

**Comment by barachka** [comment]
Please, accept request.

**Comment by feketemihai** [comment]
@guewen @yvaucher Code looks ok, i didn't have time to test it.

**Comment by StefanRijnhart** [comment]
Looks good, just some smaller issues:

README.md has been renamed to README.rst. Please resolve the conflict by moving your modifications there.

Querying the service fails silently if the xmltodict library is not available, as the import error is caught in a try block. While it is logged, maybe you can mention this requirement in the README?

Next, I get KeyError(u'LTL',) in the process. Lithuania has switched to the Euro, so it needs to be removed from the list of supported currencies.

**Comment by pedrobaeza** [comment]
As there's no answer for comments since a lot, I close. If there's interest, please reopen.

**Pull request closed without merging**